circuits-circom/regexes/paylah/paylah_send_amount.circom

```diff
@@ -0,0 +1,330 @@
+pragma circom 2.1.5;
+
+include "@zk-email/circuits/regexes/regex_helpers.circom";
+
+template PaylahSendAmountRegex(msg_bytes) {
+	signal input msg[msg_bytes];
+	signal output out;
+
+	var num_bytes = msg_bytes+1;
+	signal in[num_bytes];
+	in[0]<==255;
+	for (var i = 0; i < msg_bytes; i++) {
+		in[i+1] <== msg[i];
+	}
+
+	component eq[27][num_bytes];
+	component and[34][num_bytes];
+	component multi_or[3][num_bytes];
+	signal states[num_bytes+1][34];
+	component state_changed[num_bytes];
+
+	states[0][0] <== 1;
+	for (var i = 1; i < 34; i++) {
+		states[0][i] <== 0;
+	}
+
+	for (var i = 0; i < num_bytes; i++) {
+		state_changed[i] = MultiOR(33);
+		eq[0][i] = IsEqual();
+		eq[0][i].in[0] <== in[i];
+		eq[0][i].in[1] <== 46;
+		eq[1][i] = IsEqual();
+		eq[1][i].in[0] <== in[i];
+		eq[1][i].in[1] <== 48;
+		eq[2][i] = IsEqual();
+		eq[2][i].in[0] <== in[i];
+		eq[2][i].in[1] <== 49;
+		eq[3][i] = IsEqual();
+		eq[3][i].in[0] <== in[i];
+		eq[3][i].in[1] <== 50;
+		eq[4][i] = IsEqual();
+		eq[4][i].in[0] <== in[i];
+		eq[4][i].in[1] <== 51;
+		eq[5][i] = IsEqual();
+		eq[5][i].in[0] <== in[i];
+		eq[5][i].in[1] <== 52;
+		eq[6][i] = IsEqual();
+		eq[6][i].in[0] <== in[i];
+		eq[6][i].in[1] <== 53;
+		eq[7][i] = IsEqual();
+		eq[7][i].in[0] <== in[i];
+		eq[7][i].in[1] <== 54;
+		eq[8][i] = IsEqual();
+		eq[8][i].in[0] <== in[i];
+		eq[8][i].in[1] <== 55;
+		eq[9][i] = IsEqual();
+		eq[9][i].in[0] <== in[i];
+		eq[9][i].in[1] <== 56;
+		eq[10][i] = IsEqual();
+		eq[10][i].in[0] <== in[i];
+		eq[10][i].in[1] <== 57;
+		and[0][i] = AND();
+		and[0][i].a <== states[i][1];
+		multi_or[0][i] = MultiOR(11);
+		multi_or[0][i].in[0] <== eq[0][i].out;
```

Comment:
Can optimize multi_ors

Reply:
Think it's fine. It's already so few constraints.